google/cloud/internal/retry_loop_helpers.cc

```diff
@@ -110,11 +110,23 @@ StatusOr<std::chrono::milliseconds> Backoff(Status const& status,
                                             char const* location,
                                             RetryPolicy& retry,
                                             BackoffPolicy& backoff,
-                                            Idempotency idempotency) {
+                                            Idempotency idempotency,
+                                            bool enable_server_retries) {
+  bool should_retry = retry.OnFailure(status);
+  if (enable_server_retries) {
+    auto ri = internal::GetRetryInfo(status);
+    if (ri.has_value()) {
```

Comment:
Did we discuss eschewing `optional::has_value()` in explicit boolean contexts (i.e., just `if (ri)`)?  I thought so, but can't find any trace.